### ISSUE TITLE
/depsolve error: There was a problem depsolving []: [u'Success - empty transaction']

### ISSUE THREAD

**Issue by martinpitt** [rhinstaller/lorax#290]
This happens when running welder-web against current lorax-composer 19.7.5:

  * Start with no recipes; open JS console to see error messages
  * Create a new recipe; I called it "test"
  * Several errors are thrown:
    `Error fetching recipe: TypeError: null is not iterable!`
    `Error in fetchModalRecipeContents promise: undefined`
    `uncaught at fetchRecipeContents TypeError: Cannot read property 'id' of undefined`

AFAICS this happens during calling
```
❱❱❱ curl http://localhost:4000/api/v0/recipes/depsolve/test
{
  "errors": [
    {
      "msg": "There was a problem depsolving []: [u'Success - empty transaction']", 
      "recipe": "test"
    }
  ], 
  "recipes": [
    {
      "dependencies": [], 
      "modules": [], 
      "recipe": {
        "description": "foo", 
        "modules": null, 
        "name": "test", 
        "packages": null, 
        "version": "0.0.1"
      }
    }
  ]
}
```

In the systemd journal I see the same error:

    lorax-composer[1206]: 2017-12-12 03:17:30,521: (v0_recipes_depsolve) There was a problem depsolving []: [u'Success - empty transaction']

**Comment by martinpitt** [comment]
The `null is not iterable` probably refers to the fact that the bdcs-rs API returned an empty list for `packages` instead of `null`:

❱❱❱ curl http://localhost:4000/api/v0/recipes/depsolve/test
{"recipes":[{"recipe":{"name":"test","description":"foo","version":"0.0.1","modules":[],"packages":[]},"modules":[],"dependencies":[]}]}

This should arguably be fixed in welder-web itself to become more tolerant, but for achieving better compatibility it would be nice to always return a list in lorax-composer.

**Comment by bcl** [comment]
Fixed by #291
